tests: subsys: Check zephyr os tracing functions.

Zhaoningx opened the pull request:
Add some testcases for subsys/trace and Use
self-define backends to get tracing data
and check the output data.

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

Commit:
tests: subsys: Check zephyr os tracing functions.

Add some testcases for subsys/trace and Use
self-define backends to get tracing data
and check the output data.

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

enjiamai commented on tests/subsys/tracing/tracing_sys/src/main.c:
```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
```
Please change the year to 2021. So does  "tests/subsys/tracing/tracing_sys/src/tracing_string_format_test.c: line 2"

Zhaoningx replied:
Done, thanks

enjiamai commented on tests/subsys/tracing/tracing_sys/src/main.c:
```diff
@@ -0,0 +1,166 @@
+ *
+ * Assumptions and Constraints:
+ * - N/A
+ */
```
Excuse me, could you please remove this kind of comment at this moment? Just keep good "brief, detail, ingroup, see, etc" in place.

enjiamai commented on tests/subsys/tracing/tracing_sys/src/main.c:
```diff
@@ -0,0 +1,166 @@
+ */
+#include <ztest.h>
+#if defined(CONFIG_TRACING_BACKEND_UART) || defined(CONFIG_TRACING_BACKEND_USB)
+#include "../../../../subsys/tracing/include/tracing_backend.h"
```
Can we make it shorter, just like "include/tracing_backend.h" or something like this?

Zhaoningx replied:
It doesn't work if I check it, maybe need some time to take a look at this, thanks.

LeiW000 commented on tests/subsys/tracing/tracing_sys/src/main.c:
```diff
@@ -0,0 +1,136 @@
+#ifdef CONFIG_TRACING_CTF
+	ztest_test_skip();
+#endif
+	generate_more_tracing_data();
```
Here, we need to make sure that "thread" already exits.

LeiW000 replied:
Have you addressed this comment with the latest commit?

Zhaoningx replied:
Hi @LeiW000 , I added a API k_thread_join() in the funciton generate_more_tracing_data, and the sub-thread has already  exited before getting here. I don't know if I fixed this comment.

Zhaoningx commented:
Hi @nashif , would you like to take a look at this PR, thanks.

Zhaoningx commented:
Hi @nashif @andyross , I think I have addressed all comments, and the testcase is used to test tracing function by using the self-designed tracing backend on qemu_x86 platform, and you know, zephyr branch will be released on date 5.7, so I want to got merged before this deadline, so would you like to take a look at this PR, and to comment it if there are anything wrong, thanks very much.

nashif commented:
marking as DNM, we expect some major changes to tracing subsystems coming from #33603, will revisit and integrate with new changes once they land.
